megamek/src/megamek/common/pathfinder/BoardEdgePathFinder.java

```diff
@@ -94,34 +102,196 @@ else if(entity.getPosition().getY() <= (board.getHeight() / 2) &&
      * @param entity The entity
      * @param destinationRegion The region
      */
-    private void setAppropriateFacing(Entity entity, int destinationRegion) {
+    private int getAppropriateFacing(Entity entity, int destinationRegion) {
         switch(destinationRegion) {
         case Board.START_N:
-            entity.setFacing(0);
-            break;
+            return 0;
         case Board.START_S:
-            entity.setFacing(3);
-            break;
+            return 3;
         case Board.START_E:
             if(entity.getPosition().getY() < entity.getGame().getBoard().getHeight() / 2) {
-                entity.setFacing(2);
+                return 2;
             }
             else {
-                entity.setFacing(1);
+                return 1;
             }
-            break;
         case Board.START_W:
             if(entity.getPosition().getY() < entity.getGame().getBoard().getHeight() / 2) {
-                entity.setFacing(4);
+                return 4;
             }
             else {
-                entity.setFacing(5);
+                return 5;
             }
-            break;
         }
+        
+        return -1;
+    }
+    
+    /**
+     * Helper function to directly the entity to an appropriate facing given the destination region
+     * Changes the actual entity's facing.
+     * @param entity The entity
+     * @param destinationRegion The region
+     */
+    private void setAppropriateFacing(Entity entity, int destinationRegion) {
+        entity.setFacing(getAppropriateFacing(entity, destinationRegion));
     }
     
     /**
+     * Helper method that attempts to find a path that connects from the entity's current position
+     * to the path's desired edge. The reason being that a particular path may technically lead to an edge, 
+     * but we cut the path generation short when it reaches another path that already goes to that edge.
+     * @param startingPath
+     * @return
+     */
+    public MovePath findCombinedPath(Entity entity) {
```

Comment:
This, "joinPaths" and "matchFacingToPaths" methods are currently unused. The intent is to eventually be able to generate an uninterrupted valid path to the desired edge, for such situations as a unit being under forced withdrawal or having no targets visible/in LOS.